platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java

```diff
@@ -16,46 +16,112 @@
-            @NonNull final String memo)
-            throws ParseException {
-        Objects.requireNonNull(addressLine, "The addressLine must not be null.");
-        Objects.requireNonNull(id, "The id must not be null.");
-        Objects.requireNonNull(isOwnHostDeterminer, "The isOwnHostDeterminer must not be null.");
-        Objects.requireNonNull(memo, "The memo must not be null.");
-        final String[] parts = addressLine.trim().split(",");
-        if (parts.length < 8 || parts.length > 9) {
+    public static String addressBookConfigText(@NonNull final AddressBook addressBook) {
+        Objects.requireNonNull(addressBook, "The addressBook must not be null.");
+        final TextTable table = new TextTable().setBordersEnabled(false);
+        for (final Address address : addressBook) {
+            final String memo = address.getMemo();
+            final boolean hasMemo = !memo.trim().isEmpty();
+            final boolean hasInternalIpv4 = address.getAddressInternalIpv4() != null;
+            final boolean hasExternalIpv4 = address.getAddressExternalIpv4() != null;
+            table.addRow(
+                    "address,",
+                    address.getNodeId() + ",",
+                    address.getNickname() + ",",
+                    address.getSelfName() + ",",
+                    address.getWeight() + ",",
+                    (hasInternalIpv4 ? ipString(address.getAddressInternalIpv4()) : "") + ",",
+                    address.getPortInternalIpv4() + ",",
+                    (hasExternalIpv4 ? ipString(address.getAddressExternalIpv4()) : "") + ",",
+                    address.getPortExternalIpv4() + (hasMemo ? "," : ""),
+                    memo);
+        }
+        return table.render();
+    }
+
+    /**
+     * Parses an address book from text in the form described by config.txt.  Comments are ignored.
+     *
+     * @param addressBookText the config.txt compatible serialized address book to parse.
+     * @return a parsed AddressBook.
+     * @throws ParseException if any Address throws a ParseException when being parsed.
+     */
+    @NonNull
+    public static AddressBook parseAddressBookText(@NonNull final String addressBookText) throws ParseException {
+        Objects.requireNonNull(addressBookText, "The addressBookText must not be null.");
+        final AddressBook addressBook = new AddressBook();
+        for (final String addressLine : addressBookText.split("\\r?\\n")) {
+            final Address address = parseAddressText(addressLine);
+            if (address != null) {
+                addressBook.add(address);
+            }
+        }
+        return addressBook;
+    }
+
+    /**
+     * Parse an address from a single line of text, if it exists.  Address lines may have comments which start with the
+     * `#` character.  Comments are ignored.  Lines which are just comments return null.  If there is content prior to a
```

Comment:
Why do you always do 2 spaces between sentences?

Reply:
I am 43 and learned to type pre-internet.  The 2 spaces is automatic.  When text was typed on type writers, it was the standard to add two spaces after periods.  HTML rendering eliminated the double spacing and collapsed all white space.  This caused the standard to change.  I just haven't adapted since.

Reply:
https://www.theatlantic.com/science/archive/2018/05/two-spaces-after-a-period/559304/

Reply:
https://www.masterclass.com/articles/how-many-spaces-after-a-period